Automattic/jetpack
# JSON API: Add jetpack_connection_active_plugins option in site endpoint.

Adds `jetpack_connection_active_plugins` option to the options returned by the JSON API site fetching endpoints.

The concept is that `jetpack_connection_active_plugins` option should be present in not only single site fetching requests but also when fetching the list of owned sites from WPCOM (`GET me/sites`).
I believe it also make sense for this option to be present in all cases, this is why the approach here is to add it in the `after_render_options` method.

#### Changes proposed in this Pull Request:

* Adds abstract protected `get_connection_active_plugins` in `Abstract_Jetpack_Site`
* Implements the above in `Jetpack_Site` so that `jetpack_connection_active_plugins` option is present in the options returned by the JSON API site fetching endpoints (`GET site/%`).

#### Jetpack product discussion


p9dueE-38v-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:
* Create a JN site from this PR and fully connect Jetpack
* Go to the developer console and try the `GET sites/YOUR_SITE_ID_OR_DOMAIN` endpoint. Remember to test for all API versions, eg: `v1`, `v1.1`, `v1.2`
* Confirm that the `jetpack_connection_active_plugins` option is present in the `options` field.
